src/ScreenManager.cpp

```diff
@@ -262,23 +262,25 @@ ScreenManager::ScreenManager()
 
 ScreenManager::~ScreenManager()
 {
-	LOG->Trace( "ScreenManager::~ScreenManager()" );
-	LOG->UnmapLog( "ScreenManager::TopScreen" );
+	LOG->Trace("ScreenManager::~ScreenManager()");
+	LOG->UnmapLog("ScreenManager::TopScreen");
 
-	SAFE_DELETE( g_pSharedBGA );
-	for( unsigned i=0; i<g_ScreenStack.size(); i++ )
+	SAFE_DELETE(g_pSharedBGA);
+	for (auto& screen : g_ScreenStack)
```

Comment:
Try not using auto unless it's some obvious standard type that might get long (e.g. iterators). Here for example, doing `LoadedScreen& screen` isn't that much longer and also doesn't hide the type. Here and below.